pytorch_lightning/trainer/trainer.py

```diff
@@ -511,7 +511,7 @@ def _setup_on_init(self, num_sanity_val_steps: int) -> None:
 
         self.should_stop = False
         self.state = TrainerState()
-        self.num_training_batches = 0
+        self.num_training_batches = float("inf")
```

Comment:
is this inconsistent if we don't do the same for validation/test/predict batches too?